lib/open_project/costs/patches/query_patch.rb

```diff
@@ -79,7 +79,8 @@ def available_work_package_filters_with_costs
           'cost_object_id' => {
             type: :list_optional,
             order: 14,
-            values: CostObject.all(conditions: ['project_id IN (?)', project], order: 'subject ASC').map { |d| [d.subject, d.id.to_s] }
+            values: CostObject.where(['project_id IN (?)', project])
+                    .order('subject ASC').map { |d| [d.subject, d.id.to_s] }
```

Comment:
@NobodysNightmare Rails 4.0. In which case, there are probably plenty of search & replace opportunities for you in core :stuck_out_tongue: